kafka/admin/client.py

```diff
@@ -133,6 +133,20 @@ class KafkaAdminClient(object):
             Default: None
         sasl_kerberos_service_name (str): Service name to include in GSSAPI
             sasl mechanism handshake. Default: 'kafka'
+        sasl_oauth_token_provider (Object): OAuthBearer token provider instance.
```

Comment:
Rather than `object`, consider defining an abstract interface class w/ `abc`. See kafka/serializer/abstract.py as an example. Alternately, consider splitting into two separate configs, `sasl_oauth_token` and `sasl_oauth_extensions`, each being simple `callable` type

Reply:
Oh I didn't know we could specify abstract interfaces! I will do that then in a `kafka/oauth` directory